### PR TITLE
SUP-1609 Add default team attribute to pipeline resource

### PR DESCRIPTION
**Pull request by jradtilbrook**
## PR checklist:
- [x] `docs/` updated
- [ ] tests added
- [x] an example added to `examples/` (useful to demo new field/resource)
- [x] `CHANGELOG.md` updated with pending release information

### CODE REVIEW

**Comment by jradtilbrook** [comment]
We are not set up for testing this since it will need a different kind of token. I've tested it locally using a separate user that is not an admin and it looks good, it doesn't have 50 teams though but that edge case should be okay I think

File: buildkite/resource_pipeline.go [hunk 1]
```diff
@@ -745,6 +833,38 @@ func (p *pipelineResource) Update(ctx context.Context, req resource.UpdateReques
...
 
+// setDefaultTeamIfExists will try to find a team for the pipeline to set as default
+// if we got here from a terraform import, we will have no idea what (if any) team to assign as the default, it
+// will need to be done manually by the user
+// however, if this is a normal read operation, we will have access to the previous state which is a reliable
+// source of default team ID. so if it is set in state, we need to ensure the permission level is correct,
+// otherwise it cannot be the default owner if it has lower permissions
+// findAndRemoveTeam will try to find a team and remove its access from the pipeline
+// we only know the teams ID but the API request to remove access requies the pipeline team connection ID, so we need to
+// query all connected teams and check their ID matches
```
Comment (mcncl) [inline buildkite/resource_pipeline.go:844]: Looks like this is a  partly a copy/paste of the comment for the above function and can be removed/reduced.

**Review by mcncl — CHANGES_REQUESTED** [review]
Looks good. Any tests planned?

File: buildkite/resource_pipeline.go (hunk 1 above)
Comment (jradtilbrook) [reply buildkite/resource_pipeline.go:844]: Oh yeah forgot to delete that top section 👍

**Review by mcncl — DISMISSED** [review]

**Review by james2791 — APPROVED** [review]
As 🧪 ed - with the caveat as above

Love the recursive implementation - something the other resources with said cursor window lookup like cluster queue/token etc use (end state for those is a `node` query)!

**Pull request merged**